src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -0,0 +1,103 @@
+import { useState } from 'react';
+import { background, border, cn, color, pressable } from '../../styles/theme';
+import type { SwapSettingsSlippageInputReact } from '../types';
+import { useSwapContext } from './SwapProvider';
+
+export function SwapSettingsSlippageInput({
+  className,
+  defaultSlippage = 3,
+}: SwapSettingsSlippageInputReact) {
+  const { setLifeCycleStatus } = useSwapContext();
+  const [slippage, setSlippage] = useState(defaultSlippage);
+  const [isAutoMode, setIsAutoMode] = useState(true);
+
+  const updateSlippage = (newSlippage: number) => {
+    setSlippage(newSlippage);
+    setLifeCycleStatus({
+      statusName: 'slippageChange',
+      statusData: { maxSlippage: newSlippage },
+    });
+  };
+
+  const handleSlippageChange = (newSlippage: string) => {
+    const newSlippageNumber = Number.parseFloat(newSlippage);
+    if (!Number.isNaN(newSlippageNumber)) {
+      updateSlippage(newSlippageNumber);
+    }
+  };
+
+  const handleModeChange = (auto: boolean) => {
+    setIsAutoMode(auto);
+    if (auto) {
+      updateSlippage(defaultSlippage);
+    }
+  };
+
+  return (
+    <div className={className}>
```

Comment:
3 `divs`, anytime you see more then 1 div, always think, can this be more semantic?

Reply:
improving semantics to use `section`, `fieldset`, and `legend`